File: Medium/Python/2216. Minimum Deletions to Make Array Beautiful.py
class Solution(object):
    def minDeletion(self, nums):
        delete = 0

        for i in range(len(nums) - 1):
            index = delete % 2 == i % 2
            if index and nums[i] == nums[i + 1]:
                delete += 1

        if (len(nums) - delete) % 2 == 0:
            return delete
        return delete + 1

File: Medium/Java/2216. Minimum Deletions to Make Array Beautiful.java
class Solution {
    public int minDeletion(int[] nums) {
        int del = 0;

        for(int i = 0; i < nums.length - 1; i += 2){
            if(nums[i] == nums[i + 1]){
                del++;
                i--;
            }
        }

        if((nums.length - del) % 2 == 0)
            return del;
        return del + 1;
    }
}
